plugins/inputs/kube_inventory/README.md

```diff
@@ -0,0 +1,245 @@
+- Limit series cardinality in your database using the
+  [max-series-per-database][] and [max-values-per-tag][] settings.
+- Consider using the [Time Series Index][tsi].
+- Monitor your databases [series cardinality][].
+- Consult the [InfluxDB documentation][influx-docs] for the most up-to-date techniques.
+
+### Configuration:
+
+```toml
+[[inputs.kube_inventory]]
+  ## URL for the kubelet
+  url = "https://127.0.0.1"
+
+  ## Namespace to use
+  # namespace = "default"
+
+  ## Use bearer token for authorization. ('bearer_token' takes priority)
+  # bearer_token = "/path/to/bearer/token"
+  ## OR
+  # bearer_token_string = "abc_123"
+
+  ## Set response_timeout (default 5 seconds)
+  # response_timeout = "5s"
+
+  ## Optional Resources to exclude from gathering
+  ## Leave them with blank with try to gather everything available.
+  ## Values can be - "daemonsets", deployments", "nodes", "persistentvolumes",
+  ## "persistentvolumeclaims", "pods", "statefulsets"
+  # resource_exclude = [ "deployments", "nodes", "statefulsets" ]
+
+  ## Optional Resources to include when gathering
+  ## Overrides resource_exclude if both set.
+  # resource_include = [ "deployments", "nodes", "statefulsets" ]
+
+  ## Optional max age for config map
+  # max_config_map_age = "1h"
+
+  ## Optional TLS Config
+  # tls_ca = "/path/to/cafile"
+  # tls_cert = "/path/to/certfile"
+  # tls_key = "/path/to/keyfile"
+  ## Use TLS but skip chain & host verification
+  # insecure_skip_verify = false
+```
+
+#### Kubernetes Permissions
+
+If using [RBAC authorization](https://kubernetes.io/docs/reference/access-authn-authz/rbac/), you will need to create a cluster role to list "persistentvolumes" and "nodes". You will then need to make an [aggregated ClusterRole](https://kubernetes.io/docs/reference/access-authn-authz/rbac/#aggregated-clusterroles) that will eventually be bound to a user or group.
+```yaml
+---
+kind: ClusterRole
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: influx:cluster:viewer
+  labels:
+    rbac.authorization.k8s.io/aggregate-view-telegraf: "true"
+rules:
+- apiGroups: [""]
+  resources: ["persistentvolumes","nodes"]
+  verbs: ["get","list"]
+
+---
+kind: ClusterRole
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: influx:telegraf
+aggregationRule:
+  clusterRoleSelectors:
+  - matchLabels:
+      rbac.authorization.k8s.io/aggregate-view-telegraf: "true"
+      rbac.authorization.k8s.io/aggregate-to-view: "true"
+rules: [] # Rules are automatically filled in by the controller manager.
+```
+
+Bind the newly created aggregated ClusterRole with the following config file, updating the subjects as needed.
+```yaml
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: influx:telegraf:viewer
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: influx:telegraf
+subjects:
+- kind: ServiceAccount
+  name: telegraf
+  namespace: default
```

Comment:
I think that after explaining how to setup the permissions here we should have an example on how to deploy telegraf for this specific use case using a deployment with a configmap configured to use this plugin. That's what I would expect to get from this specific plugin documentation when reading it.

Reply:
Is that something you can provide?

Reply:
Yup will do :)